code/parachain/runtime/common/src/fees.rs

```diff
@@ -47,11 +48,16 @@ pub fn multi_existential_deposits<AssetsRegistry>(_currency_id: &CurrencyId) ->
 }
 
 #[cfg(not(feature = "runtime-benchmarks"))]
-pub fn multi_existential_deposits<AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>>(
+pub fn multi_existential_deposits<
+	AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>
+		+ AssetExistentialDepositInspect<AssetId = CurrencyId, Balance = Balance>,
+>(
 	currency_id: &CurrencyId,
 ) -> Balance {
-	PriceConverter::<AssetsRegistry>::to_asset_balance(NATIVE_EXISTENTIAL_DEPOSIT, *currency_id)
-		.unwrap_or(Balance::MAX)
+	AssetsRegistry::existential_deposit(*currency_id).unwrap_or_else(|_| {
+		PriceConverter::<AssetsRegistry>::to_asset_balance(NATIVE_EXISTENTIAL_DEPOSIT, *currency_id)
+			.unwrap_or(Balance::MAX)
+	})
 }
 
 pub struct PriceConverter<AssetsRegistry>(PhantomData<AssetsRegistry>);
```

Comment:
in this file there is WellKnownForeignToNativePriceConverter struct that has ratios and existential deposits hard coded. are we going to remove this code? it is used in assets pallet rpc but i think we will remove assets from runtime?

Reply:
Assuming a successful migration, this will not have value.
Will remove and implement `PriceConverter` on Assets registry.

Reply:
e1b9e551a7646bddc53125f2195aaed890e0e5ca